src/app/shared/component/consumption-chart/consumption-chart.component.ts

```diff
@@ -503,58 +550,19 @@ export class ConsumptionChartComponent implements AfterViewInit {
           onClick: (e, legendItem, legend) => {
             const dataset = legend.chart.data.datasets[legendItem.datasetIndex];
             const status = dataset.hidden;
             dataset.hidden = !status;
             this.data.datasets[legendItem.datasetIndex].hidden = !status;
-            const visibleDatasets = this.data.datasets.filter(ds => !ds.hidden).map(ds => ds.order);
-            for (const key in this.gridDisplay) {
-              if(Object.prototype.hasOwnProperty.call(this.gridDisplay, key)){
-                this.gridDisplay[key] = false;
-              }
-            }
-            if (
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.INSTANT_WATTS) ||
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.INSTANT_WATTS_L1) ||
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.INSTANT_WATTS_L2) ||
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.INSTANT_WATTS_L3) ||
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.CUMULATED_CONSUMPTION_WH) ||
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.LIMIT_WATTS)
-            ) {
-              this.gridDisplay[ConsumptionChartAxis.POWER] = true;
-            } else if (
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.INSTANT_AMPS) ||
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.INSTANT_AMPS_L1) ||
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.INSTANT_AMPS_L2) ||
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.INSTANT_AMPS_L3) ||
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.INSTANT_AMPS_DC) ||
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.CUMULATED_CONSUMPTION_AMPS) ||
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.LIMIT_AMPS)
-            ) {
-              this.gridDisplay[ConsumptionChartAxis.AMPERAGE] = true;
-            } else if (
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.INSTANT_VOLTS) ||
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.INSTANT_VOLTS_L1) ||
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.INSTANT_VOLTS_L2) ||
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.INSTANT_VOLTS_L3) ||
-              visibleDatasets.includes(ConsumptionChartDatasetOrder.INSTANT_VOLTS_DC)
-            ) {
-              this.gridDisplay[ConsumptionChartAxis.VOLTAGE] = true;
-            } else if( visibleDatasets.includes(ConsumptionChartDatasetOrder.STATE_OF_CHARGE) ) {
-              this.gridDisplay[ConsumptionChartAxis.PERCENTAGE] = true;
-            } else if( visibleDatasets.includes(ConsumptionChartDatasetOrder.CUMULATED_AMOUNT) ) {
-              this.gridDisplay[ConsumptionChartAxis.AMOUNT] = true;
-            }
+            this.updateVisibleGridLines();
             legend.chart.options = this.createOptions();
```

Comment:
We need to call createOptions again because the grid lines in the background are dynamically generated based on the visible legends. In scales option, we are setting the drawOnChartArea based on gridDisplay array values. This is updated when a legend is made visible or hidden. Technically, we might be able to update only the scales instead. 
To see the side effect on hiding this change, comment this line and open a sessions chart. Hide all fields that use (kW) or (kW.h). The background horizontal grid lines will disappear.